scripts/ci/classify_pr_scope.py

```
@@ -0,0 +1,93 @@
+def normalize_path(raw_path: str) -> str:
+    return raw_path.strip().replace("\\", "/").lstrip("./")
+
+
+def matches_any(path: str, patterns: list[str]) -> bool:
+    pure_path = PurePosixPath(path)
+    return any(pure_path.match(pattern) or fnmatch.fnmatch(path, pattern) for pattern in patterns)
+
+
+def changed_paths(repo_root: Path, base: str, head: str) -> list[str]:
+    proc = subprocess.run(
+        ["git", "-C", str(repo_root), "diff", "--name-only", "--diff-filter=ACMR", f"{base}...{head}"],
```

Ruff (0.15.4)

[error] Starting a process with a partial executable path

(S607)